### PR TITLE
Extend functionality of add and subtract operator

### PR DESCRIPTION
**Pull request by Zainab-Saad**

### CODE REVIEW

**Comment by Zainab-Saad** [comment]
@MuhammadTahaNaveed any updates on this one?

**Comment by Zainab-Saad** [comment]
@MuhammadTahaNaveed 
Fixed the issue with sub operator

**Comment by MuhammadTahaNaveed** [comment]
@Zainab-Saad 
```
SELECT '{"a":1 , "b":2, "c":3}'::agtype - null;
SELECT '["a","b","c"]'::agtype - null;
```
Output is agtype null for these cases, shouldnt it be lhs array/object?

File: regress/expected/agtype.out [hunk 1]
```diff
@@ -692,6 +704,475 @@ SELECT '3.14::numeric'::agtype + '3.14::numeric'::agtype;
...
+(1 row)
+
+SELECT '{"id": 1688849860263951, "label": "e_var", "end_id": 281474976710664, "start_id": 281474976710663, "properties": {"id": 0}}::edge'::agtype + '{"id": 1125899906842625, "label": "Vertex", "properties": {"a": "xyz", "b": true, "c": -19.888}}::vertex'::agtype;
+                                                                                                                   ?column?                                                                                                                    
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ [{"id": 1688849860263951, "label": "e_var", "end_id": 281474976710664, "start_id": 281474976710663, "properties": {"id": 0}}::edge, {"id": 1125899906842625, "label": "Vertex", "properties": {"a": "xyz", "b": true, "c": -19.888}}::vertex]
+(1 row)
+
+-- errors out
+SELECT '{"a":1 , "b":2, "c":3}'::agtype + '{"b": 2}';
```
Comment (MuhammadTahaNaveed) [inline regress/expected/agtype.out:1170]: Shouldnt this work ?

**Comment by Zainab-Saad** [comment]
@MuhammadTahaNaveed 
```
SELECT '["a","b","c"]'::agtype - 'null'::agtype;
ERROR:  expected agtype integer, not agtype NULL
```

`SELECT '["a","b","c"]'::agtype - null;` does not hit the `agtype_sub` breakpoint

Same with agtype object on lhs

PS: I think that is the case with other ops

File: regress/expected/agtype.out (hunk 1 above)
Comment (Zainab-Saad) [reply regress/expected/agtype.out:1170]: With this output ` {"a": 1, "b": 2, "c": 3}`?

Comment (MuhammadTahaNaveed) [reply regress/expected/agtype.out:1170]: yes I guess. For non existent key, it would extend.

**Pull request merged**